src/wp-includes/script-loader.php

```diff
@@ -3027,3 +3027,522 @@ function wp_enqueue_block_style( $block_name, $args ) {
 	// Enqueue assets in the editor.
 	add_action( 'enqueue_block_assets', $callback );
 }
+
+/**
+ * Runs the theme.json webfonts handler.
+ *
+ * Using `WP_Theme_JSON_Resolver`, it gets the fonts defined
+ * in the `theme.json` for the current selection and style
+ * variations, validates the font-face properties, generates
+ * the '@font-face' style declarations, and then enqueues the
+ * styles for both the editor and front-end.
+ *
+ * Design Notes:
+ * This is not a public API, but rather an internal handler.
+ * A future public Webfonts API will replace this stopgap code.
+ *
+ * This code design is intentional.
+ *    a. It hides the inner-workings.
+ *    b. It does not expose API ins or outs for consumption.
+ *    c. It only works with a theme's `theme.json`.
+ *
+ * Why?
+ *    a. To avoid backwards-compatibility issues when
+ *       the Webfonts API is introduced in Core.
+ *    b. To make `fontFace` declarations in `theme.json` work.
+ *
+ * @link  https://github.com/WordPress/gutenberg/issues/40472
+ *
+ * @since 6.0.0
+ */
+function _wp_theme_json_webfonts_handler() {
+	// Webfonts to be processed.
+	$registered_webfonts = array();
+
+	/**
+	 * Gets the webfonts from theme.json.
+	 *
+	 * @since 6.0.0
+	 *
+	 * @return array Array of defined webfonts.
+	 */
+	$fn_get_webfonts_from_theme_json = static function() {
+		// Get settings from theme.json.
+		$settings = WP_Theme_JSON_Resolver::get_merged_data()->get_settings();
+
+		// If in the editor, add webfonts defined in variations.
+		if ( is_admin() || ( defined( 'REST_REQUEST' ) && REST_REQUEST ) ) {
+			$variations = WP_Theme_JSON_Resolver::get_style_variations();
+			foreach ( $variations as $variation ) {
+				// Skip if fontFamilies are not defined in the variation.
+				if (
+					empty( $variation['settings'] ) ||
+					empty( $variation['settings']['typography'] ) ||
+					empty( $variation['settings']['typography']['fontFamilies'] )
```

Comment:
```suggestion
					empty( $variation['settings']['typography']['fontFamilies'] )
```

`empty()` includes the first two checks are included in the third.

Reply:
Resolved in https://github.com/WordPress/wordpress-develop/pull/2622/commits/58839d593e1f59f07589b7ec26596013c1565268